Modules/_ssl.c

```diff
@@ -2397,11 +2398,23 @@ _ssl__SSLSocket_read_impl(PySSLSocket *self, int len, int group_right_1,
 
     do {
         PySSL_BEGIN_ALLOW_THREADS
-        retval = SSL_read_ex(self->ssl, mem, len, &count);
+        do {
+            retval = SSL_read_ex(self->ssl, mem + got, len, &count);
+            if(retval <= 0) {
+                break;
+            }
+
+            got += count;
+            len -= count;
+        } while(nonblocking && len > 0);
```

Comment:
``got`` and ``count`` are confusing variable names. I suggest that you keep ``count`` as counter for the ``read()`` function call and introduce a new variable for ``SSL_read_ex()`` call.

Is it a good idea to have a potentially infinite loop here that ignores signals and timeouts? This smells like DoS vunlerable to be happen.

```suggestion
            retval = SSL_read_ex(self->ssl, mem + got, len, &readbytes);
            if (retval <= 0) {
                break;
            }

            count += readbytes;
            len -= readbytes;
        } while (nonblocking && len > 0);
```